### PR TITLE
Widget Visibility: Support split terms

### PR DESCRIPTION
**Pull request by danielhomer**

### CODE REVIEW

**Comment by kraftbj** [comment]
Background: 4.2 split terms whenever a term was updated. 4.3 will split all terms upon upgrade.

**Comment by spacedmonkey** [comment]
Considering that WordPress 4.3 went out yesterday, this bug needs to be priority, as those who were running 4.3 may have broken visibility settings.

**Comment by kraftbj** [comment]
@spacedmonkey It does have priority. Updates will appear here soon.

**Comment by samhotchkiss** [comment]
We're running tests against 4.3, expect this to be in 3.7

**Comment by zinigor** [comment]
Read the changes, tested with joined terms on 4.1, then on 4.2, then on 4.3 after splitting. Everything works properly, term IDs get updated after splitting.

File: modules/widget-visibility/widget-conditions.php [hunk 1]
```diff
@@ -142,6 +146,13 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				$taxonomies = get_taxonomies( array( '_builtin' => false ), 'objects' );
 				usort( $taxonomies, array( __CLASS__, 'strcasecmp_name' ) );
 
+				$parts = explode( '_tax_', $minor );
+
+				if ( count( $parts ) === 2 ) {
```
Comment (dereksmart) [inline modules/widget-visibility/widget-conditions.php]: wp code style prefers <a href="https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions">yoda style conditionals. </a>

File: modules/widget-visibility/widget-conditions.php [hunk 2]
```diff
@@ -443,24 +454,29 @@ public static function filter_widget( $instance ) {
...
-							$tag = get_tag( $rule['minor'] );
-
-							if ( $tag && is_tag( $tag->slug ) )
+						} else {
+							$rule['minor'] = self::maybe_get_split_term( $rule['minor'], $rule['major'] );
+							if ( is_singular() && $rule['minor'] && has_tag( $rule['minor'] ) ) {
 								$condition_result = true;
+							} else {
+								$tag = get_tag( $rule['minor'] );
+								if ( $tag && is_tag( $tag->slug ) )
```
Comment (dereksmart) [inline modules/widget-visibility/widget-conditions.php]: can you add some <a href="https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#brace-style">braces to the conditionals</a>?

File: modules/widget-visibility/widget-conditions.php [hunk 3]
```diff
@@ -531,6 +549,20 @@ public static function filter_widget( $instance ) {
 	public static function strcasecmp_name( $a, $b ) {
 		return strcasecmp( $a->name, $b->name );
 	}
+
+	public static function maybe_get_split_term( $old_term_id = '', $taxonomy = '' ) {
+		$term_id = $old_term_id;
+
+		if ( $taxonomy == 'tag' ) {
```
Comment (dereksmart) [inline modules/widget-visibility/widget-conditions.php]: Yoda styles here as well please

**Comment by dereksmart** [comment]
Code looks super solid, just a couple nitpicky code style requests.  @danielhomer do you think you can get to those today?  We would like to get this into our beta which is going out today.

**Comment by dereksmart** [comment]
@danielhomer super quick, thank you for the patch! 

:shipit:

**Comment by danielhomer** [comment]
@dereksmart no problem :)

**Pull request merged**